docs/src/main/asciidoc/vault.adoc

```diff
@@ -492,6 +492,45 @@ You should see:
 {password=connor}
 ----
 
+[[totp]]
+== TOTP Secrets Engine
+
+The TOTP secrets engine generates time-based credentials according to the TOTP standard.
+
+Vault supports both the generator scenario (like Google Authenticator) and the provider scenario (like the Google.com sign in).
+You can read more about Vault TOTP at https://www.vaultproject.io/docs/secrets/totp/.
+
+The Vault extension also supports TOTP by integrating with TOTP secret engine by providing an `io.quarkus.vault.VaultTOTPSecretEngine` class.
```

Comment:
`also` can be dropped: `The Vault extension integrates with the http://...[Vault TOTP] secret engine by providing...`